gravitee-apim-console-webui/src/management/api/api-traffic-v4/webhook-logs-details/webhook-logs-details.component.ts

```diff
@@ -150,72 +213,122 @@ export class WebhookLogsDetailsComponent implements OnInit {
       return;
     }
 
+    const lastError = log.additionalMetrics[LAST_ERROR_KEY];
+    const retryCount = log.additionalMetrics['int_webhook_retry-count'];
+    const reqHeaders = log.additionalMetrics[REQ_HEADERS_KEY];
+    const respHeaders = log.additionalMetrics[RESP_HEADERS_KEY];
+    const reqBody = log.additionalMetrics[REQ_BODY_KEY];
+    const respBody = log.additionalMetrics[RESP_BODY_KEY];
+
+    if (log.status === 0) {
+      this.connectionFailure = {
+        errorKey: log.errorKey ?? null,
+        componentType: log.errorComponentType ?? null,
+        componentName: log.errorComponentName ?? null,
+        message: log.message ?? null,
+        diagnostics: log.warnings ?? [],
+        lastError: typeof lastError === 'string' ? lastError : null,
+      };
+    } else {
+      this.connectionFailure = null;
+    }
+
     this.deliveryAttemptsDataSource = this.buildDeliveryAttempts(log);
-    const attemptsCount = this.deliveryAttemptsDataSource.length || (log.additionalMetrics?.['int_webhook_retry-count'] ?? 1);
+    const attemptsCount = this.deliveryAttemptsDataSource.length || (isNumber(retryCount) ? retryCount : 1);
 
     this.overviewRequest = [
-      { label: 'Date', value: this.formatDate(log.timestamp) },
+      { label: 'Date', value: this.formatDateTime(log.timestamp) },
       { label: 'Method', value: log.method },
       { label: 'Attempts', value: attemptsCount },
       { label: 'Callback URL', value: log.callbackUrl ?? log.uri },
     ];
 
     this.overviewResponse = [
       { label: 'Status', value: log.status },
-      { label: 'Total duration', value: log.duration || `${(log.gatewayResponseTime / 1000).toFixed(1)} s` },
+      {
+        label: 'Total duration',
+        value: log.duration || `${(log.gatewayResponseTime / 1000).toFixed(1)} s`,
+      },
       { label: 'Payload size', value: this.formatPayloadSize(log) },
-      { label: 'Last error', value: log.additionalMetrics?.['string_webhook_last-error'] ?? '—' },
+      { label: 'Last error', value: typeof lastError === 'string' ? lastError : '—' },
     ];
 
-    this.requestHeaders = this.parseHeaders(log.additionalMetrics?.['json_webhook_req-headers']);
-    this.responseHeaders = this.parseHeaders(log.additionalMetrics?.['json_webhook_resp-headers']);
-    this.requestBody = log.additionalMetrics?.['string_webhook_req-body'] ?? '';
-    this.responseBody = log.additionalMetrics?.['string_webhook_resp-body'] ?? '';
-    this.connectionFailure = this.buildConnectionFailure(log);
+    this.requestHeaders = this.parseHeaders(typeof reqHeaders === 'string' ? reqHeaders : null);
+
+    this.responseHeaders = this.parseHeaders(typeof respHeaders === 'string' ? respHeaders : null);
+    this.requestBody = typeof reqBody === 'string' ? reqBody : '';
+
+    this.responseBody = typeof respBody === 'string' ? respBody : '';
```

Comment:
Split request and response attribut together to make it easier to read

```suggestion
     this.requestHeaders = this.parseHeaders(typeof reqHeaders === 'string' ? reqHeaders : undefined);
     this.requestBody = typeof reqBody === 'string' ? reqBody : '';

    this.responseHeaders = this.parseHeaders(typeof respHeaders === 'string' ? respHeaders : undefined);
    this.responseBody = typeof respBody === 'string' ? respBody : '';
    
   You could also extract it in a private function to respect the DRY principle:
   
   private setHeader(header: <string | type>): <string> | undefined{
    return this.parseHeaders(typeof header === 'string' ? header : undefined);
   }
   And then :
   
   this.requestHeaders = this.setHeader(reqHeaders);
   this.requestBody = this.setBody(reqBody);
   
   
```

Reply:
Done! [fcbd571](https://github.com/gravitee-io/gravitee-api-management/pull/14303/commits/fcbd571882cefbb6a80b68352efc6ac594d43cd6)